tools/exportNotebook

```diff
@@ -93,33 +283,96 @@ then
         ln -s ${tmpfolder}/background_original.pdf ${tmpfolder}/background.pdf
     fi
 else
-    # Getting template files
-    sort -u "${tmpfolder}"/*.pagedata | while read -r tpl; do
-    scp $control_options -q ${SSH_IP}:"'/usr/share/remarkable/templates/${tpl}.png'" "${tmpfolder}"/
-    done
 
     # Generate a PDF file out of the templates
     sed -e "s|^|\"${tmpfolder}\"/\"|" -e 's|$|.png"|' "${tmpfolder}"/*.pagedata | tr '\n' ' ' | sed -e "s|$|-transparent white \"${tmpfolder}\"/background.pdf|" | xargs convert
+    [ ! $? -eq 0 ] && echo "Failed to generate a PDF file out of the templates" && exit 1
 fi
 
 # Extract annotations and create a PDF
-$rM2svg_cmd --input "${tmpfolder}"/*.lines --output "${tmpfolder}/foreground" $2
+IPAGE=0
+jq '.pages' "${tmpfolder}"/*.content | jq '.[]' | while read -r page; do
+    SPAGE="${page:1:${#page}-2}"
+    let IPAGE=IPAGE+1
+    if [ -f "${tmpfolder}"/*/"${SPAGE}.rm" ]; then
+        echo "Processing page $IPAGE ($SPAGE)"
+        $rM2svg_cmd --input "${tmpfolder}"/*/"${SPAGE}.rm" --output "${tmpfolder}/foreground$(printf %03d $IPAGE).svg"
+        [ ! $? -eq 0 ] && echo "Failed to generate '${tmpfolder}/foreground$(printf %03d $IPAGE).svg'" && exit 1
+    else
+        echo "Generating empty page $IPAGE ($SPAGE)"
+cat << 'EOF' > "${tmpfolder}/foreground$(printf %03d $IPAGE).svg"
+<svg xmlns="http://www.w3.org/2000/svg" height="1872" width="1404">
+        <script type="application/ecmascript"> <![CDATA[
+            var visiblePage = 'p1';
+            function goToPage(page) {
+                document.getElementById(visiblePage).setAttribute('style', 'display: none');
+                document.getElementById(page).setAttribute('style', 'display: inline');
+                visiblePage = page;
+            }
+        ]]> </script>
+    <g id="p1" style="display:inline">
+<rect x="0" y="0" width="1404" height="1872" fill-opacity="0"/></g></svg>
+EOF
+    fi
+done
+
+# Preprocess foreground*.svg
+for f in "${tmpfolder}"/foreground*.svg; do
+    # debug
+    #cp -v "$f" ~/debug/"$(basename "$f")" 
+    sed -i "s/stroke-width\:[\.0-9]\+/stroke-width:${STROKE_WIDTH}/g;s/opacity\:[\.0-9]\+/opacity:${OPACITY}/g" "$f"
+    [ ! $? -eq 0 ] && echo "Failed to reset stroke-width and opacity in '$f'" && exit 1
+done
+# - optimize svg with svgo
+if command -v "svgo" > /dev/null; then
+    for f in "${tmpfolder}"/foreground*.svg; do
+        svgo -i "$f"
+        [ ! $? -eq 0 ] && echo "Failed to optimize svg '$f'" && exit 1
```

Comment:
Is it bad if svgo fails to optimise the svg file? Could the for loop continue and leave that file (and all other failing ones) unoptimised?
```suggestion
        [ ! $? -eq 0 ] && echo "Failed to optimize svg '$f'" && continue
```